### DISCUSSION TITLE
using termux-emulator in my andoroid app for quiz and tests

### DISCUSSION THREAD

**Discussion by EnriqueStrange** [termux/termux-app#2627] [Q&A] [answered]
How to I add termux to my android app without adding the whole of termux gradels and files

**Comment by agnostic-apollo** [comment, marked as answer]
The `terminal-view` and `terminal-emulator` are libraries and need to be called by other components, like `termux-app/app`. Depending on what you want to support in your app, you will need to add most of termux `app` files and copy code from `AndroidManifest.xml` and `build.gradle` and make package name changes. It's not really simple considering termux code base is pretty huge at this point.

You can check https://github.com/simbadMarino/dCloud, it is using `updated-flavor` branch as discussed in #2441.

Or https://github.com/jackpal/Android-Terminal-Emulator on which termux is based on.

Also check https://github.com/termux/termux-app#forking
